relay-server/src/actors/processor.rs

```diff
@@ -1304,6 +1304,55 @@ impl EnvelopeProcessorService {
         Ok(())
     }
 
+    fn process_replays_combine_items(
+        &self,
+        state: &mut ProcessEnvelopeState,
+    ) -> Result<(), ProcessingError> {
+        let project_state = &state.project_state;
+        let combined_envelope_items =
+            project_state.has_feature(Feature::SessionReplayCombinedEnvelopeItems);
+
+        if combined_envelope_items {
+            // If this flag is enabled, combine both items into a single item,
+            // and remove the original items.
+            // The combined Item's payload is a MsgPack map with the keys
+            // "replay_event" and "replay_recording".
+            // The values are the original payloads of the items.
+            let envelope = &mut state.envelope_mut();
+            if let Some(replay_event_item) =
+                envelope.take_item_by(|item| item.ty() == &ItemType::ReplayEvent)
+            {
+                if let Some(replay_recording_item) =
+                    envelope.take_item_by(|item| item.ty() == &ItemType::ReplayRecording)
+                {
+                    let mut data = Vec::new();
+                    let mut combined_item_payload = BTreeMap::new();
+
+                    combined_item_payload
+                        .insert("replay_event", replay_event_item.payload().to_vec());
+                    combined_item_payload
+                        .insert("replay_recording", replay_recording_item.payload().to_vec());
+
+                    if let Err(e) = rmp_serde::encode::write(&mut data, &combined_item_payload) {
```

Comment:
is this the best way to do the msgpack encoding?

Reply:
Its a perfectly fine way to do it.  We can make it better though by pre-allocating memory in our `data` buffer (`Vec::with_capacity(some_number)`).  By doing this pushing bytes into the buffer becomes essentially free.